x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -718,6 +723,10 @@ private static List<EsIndex> buildIndices(
             final InvalidMappedField invalidField = validityVerifier.apply(fieldName, types);
             // apply verification for fields belonging to index aliases
             Map<String, InvalidMappedField> invalidFieldsForAliases = getInvalidFieldsForAliases(fieldName, types, aliases);
+            // For ESQL there are scenarios where there is no field asked from field_caps and the field_caps response only contains
+            // the list of indices. To be able to still have an "indices" list properly built (even if empty), the metadata fields are
+            // accepted but not actually added to each index hierarchy.
+            var isMetadataField = allowedMetadataFields != null && allowedMetadataFields.contains(fieldName);
```

Comment:
This trick here makes the whole analysis of FieldCapabilitiesResponse to accept metadata fields (even if by design the [IndexResolver skips them](https://github.com/elastic/elasticsearch/pull/102510/files#diff-341e59a86f203e72508f62dcce2aa897af639103da85ea5671d557f33109c1a9L710)) but only to allow the list of indices to be built. This is important for [this kind of issue](https://github.com/elastic/elasticsearch/issues/102393) where an empty list of indices in this method's result [will not make the compute service happy](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java#L141).